pkg/trace/api/api.go

```diff
@@ -49,10 +48,11 @@ type HTTPReceiver struct {
 	Stats       *info.ReceiverStats
 	RateLimiter *rateLimiter
 
-	out     chan *Payload
-	conf    *config.AgentConfig
-	dynConf *sampler.DynamicConfig
-	server  *http.Server
+	out        chan *Payload
+	conf       *config.AgentConfig
+	dynConf    *sampler.DynamicConfig
+	server     *http.Server
+	bufferPool *sync.Pool
```

Comment:
I see pools usually being defined on the global scope. Any reason it's a field? Also, I don't think it needs to be a pointer...

Reply:
I usually consider it a best practice to avoid variables. But in that case it simplifies slightly the code and I don't see the harm, I'll make the change.